### PR TITLE
SfArrow

### PR DESCRIPTION
**Pull request by przspa**
# Related issue
default
arrow_short
long
arrow

# Scope of work

# Screenshots of visual changes
default
arrow3_

long arrow
arrow4_

# Checklist

- [x] I followed [composition rules](https://github.com/DivanteLtd/storefront-ui/blob/master/docs/component-rules.md) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: src/components/atoms/SfArrow/SfArrow.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+.sf-arrow {
+  position: relative;
+  display: flex;
+  background-color: transparent;
+  border: 0;
+  padding: 14px 10px 14px;
+  background-color: #F1F2F3;
+  cursor: pointer;
+  transition: padding 150ms ease-in-out;
```
Comment (leomp12) [inline src/components/atoms/SfArrow/SfArrow.scss]: What about transition for colors too?
```scss
transition: padding 150ms ease-in-out, background-color 150ms linear, color 150ms linear;
```

Comment (przemyslawspaczek) [reply src/components/atoms/SfArrow/SfArrow.scss]: arrow2_long
arrow2

I think, now looks better

File: src/components/atoms/SfArrow/SfArrow.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default {
+  name: "SfComponent"
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.js]:
```suggestion
  name: "SfArrow"
```

File: src/components/atoms/SfArrow/SfArrow.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div class="sf-arrow">
+  <button v-on="$listeners" class="sf-arrow__button" type="button">
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.html]: Why not to make button a root element?

File: src/components/atoms/SfArrow/SfArrow.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# SfComponent
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.md]: could you please document it and move this part of docs fully to storybook? https://github.com/DivanteLtd/storefront-ui/pull/133 - for rel

File: src/components/atoms/SfArrow/SfArrow.scss [hunk 5]
```diff
@@ -0,0 +1,63 @@
+@import '../../../css/variables';
+
+.sf-arrow {
+  &__button{
+    position: relative;
+    display: flex;
+    border: 0;
+    padding: 14px 10px 14px;
+    background-color: #F1F2F3;
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.scss]: this should be changable via scss var I guess

File: src/components/atoms/SfArrow/SfArrow.scss [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    cursor: pointer;
+    outline: 0;
+    transition: padding 150ms linear,
+      background-color 150ms linear,
+      box-shadow 150ms linear;
+    box-shadow: 0 5px 12px -6px rgba(29, 31, 34, 0);
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.scss]: shadows aswell

File: src/components/atoms/SfArrow/SfArrow.scss [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        fill: #FFF;
+      }
+    }
+  }
+  &__icon {
+    position: relative;
+    z-index: 1;
+    flex: 0 0 24px;
+    height: 12px;
+    fill: #1D1F22;
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.scss]: probably this also shoudl be changable via scss var

File: src/components/atoms/SfArrow/SfArrow.scss [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    padding: 14px 10px 14px;
+    background-color: #F1F2F3;
+    cursor: pointer;
+    outline: 0;
+    transition: padding 150ms linear,
+      background-color 150ms linear,
+      box-shadow 150ms linear;
+    box-shadow: 0 5px 12px -6px rgba(29, 31, 34, 0);
+    &:hover{
+      background-color: #1D1F22;
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.scss]: scss var

File: src/components/atoms/SfArrow/SfArrow.scss [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+      box-shadow: 0 5px 12px -5px rgba(29, 31, 34, 0.6);
+
+      .sf-arrow__icon{
+        fill: #FFF;
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.scss]: scss var

File: src/components/atoms/SfArrow/SfArrow.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import { shallowMount } from "@vue/test-utils";
+import SfComponent from "@/components/template/SfComponent.vue";
+
+describe("SfComponent.vue", () => {
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.ts]:
```suggestion
describe("SfArrow.vue", () => {
```

File: src/components/atoms/SfArrow/SfArrow.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  it("renders a component", () => {
+    const component = shallowMount(SfComponent);
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.ts]:
```suggestion
    const component = shallowMount(SfArrow);
```

File: src/components/atoms/SfArrow/SfArrow.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import { shallowMount } from "@vue/test-utils";
+import SfComponent from "@/components/template/SfComponent.vue";
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.ts]:
```suggestion
import SfArrow from "./SfArrow.vue";
```

File: src/components/atoms/SfArrow/SfArrow.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+describe("SfComponent.vue", () => {
+  it("renders a component", () => {
+    const component = shallowMount(SfComponent);
+    expect(component.contains(".sf-component")).toBe(true);
```
Comment (filrak) [inline src/components/atoms/SfArrow/SfArrow.ts]:
```suggestion
    expect(component.contains(".sf-arrow")).toBe(true);
```

File: src/components/atoms/SfArrow/SfArrow.html (hunk 3 above)
Comment (przspa) [reply src/components/atoms/SfArrow/SfArrow.html]: Yup, for me is it important. Because long button have different size on default and hover state.

**Comment by przspa** [comment]
@filrak I fixed all problems

**Pull request merged**